Jint/Native/Array/ArrayPrototype.cs

```diff
@@ -1273,14 +1273,14 @@ private JsValue Join(JsValue thisObject, JsValue[] arguments)
             return JsString.Empty;
         }
 
-        static string StringFromJsValue(JsValue value)
+        static string StringFromJsValue(JsValue value, JsValue thisObject)
         {
-            return value.IsNullOrUndefined()
+            return value.IsNullOrUndefined() || thisObject == value
```

Comment:
for clarity, I think this should be `ReferenceEquals(thisObject, value)`, the operator overloads can be a bit quirky